### ISSUE TITLE
acid.nvim throws an error out of the box

### ISSUE THREAD

**Issue by daveyarwood** [clojure-vim/acid.nvim#28]
My setup:

* Ubuntu 16.04
* nvim 0.3.2-dev
* tmux 2.9

The project I'm using to try out acid.nvim is https://github.com/daveyarwood/ezzmq. It's a Boot project.

I started an nREPL via `boot repl`, then opened a source file in Neovim and immediately got this error:

```
error caught in async handler '/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid:command:AcidCommand [['Require']]'
Traceback (most recent call last):
  File "/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid/__init__.py", line 138, in acid_command
    command.call(self, self.context(), *args)
  File "/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid/commands/__init__.py", line 142, in call
    acid.command(payload, handlers)
TypeError: command() missing 1 required positional argument: 'url'
Press ENTER or type command to continue
```

The error reappears each time I run `:e` in the buffer.

**Comment by kuchichan** [comment]
I have the same problem as above (on Arch Linux). Moreover, I am a complete beginner in clojure world (I just want to start with nvim), I do not know what I'm doing wrong:

1. `lein version` - Leiningen 2.8.3 on Java 1.8.0_192 OpenJDK 64-Bit Server VM
2.  my ~/.lein/profiles.clj looks like this:
` {user: {plugins: [[refactor-nrepl "2.3.0-SNAPSHOT"]
                                [cider/cider-nrepl "0.14.0"]] } } `
3.  I create new project with `lein new app new-cool-app` Should I run something in addition? 
4.  I open a core.clj in /src/... with nvim (0.3.1) and run a REPL inside a project directory (In different console). Did I miss some steps in clojure-neovim workflow? 
5.  `cpp` and other commands initally does not work. After run `PlugUpdate` nvim recognizes Acid Commands, nevertheless I receive the error above - or indication "No REPL". 

Probably, there is some mistake in my workflow, please correct me if I am wrong (It is my second day with clojure :) )

**Comment by hkupty** [comment]
It appears that the latest commit probably wasn't as safe as I thought.

I'm preparing to simplify the usage and development of acid by changing some knobs in the implementation, but I assume something leaked out of what I thought was a controlled scope.

I'll fix that hopefully before the end of the week.

Cheers,
Henry John Kupty

**Comment by hkupty** [comment]
The fix should be in place now. I'm sorry for the mess.

**Comment by daveyarwood** [comment]
I'm still having a similar error, it's just getting thrown at a different time.

At this point, I'm able to do the following:

1. Open a Clojure file in my project.
2. Start `boot repl` in another terminal.
3. Run `:AcidInit` and then `:AcidBootstrap` (BTW, it isn't clear to me what the difference is, or what steps need to be followed when using acid.nvim. A "quickstart" in the README and help text would be much appreciated!)

When I run `:AcidBootstrap`, I get this error:

```
error caught in async handler '/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid:command:AcidCommand [['Require']]'
Traceback (most recent call last):
  File "/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid/__init__.py", line 138, in acid_command
    command.call(self, self.context(), *args)
  File "/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid/commands/__init__.py", line 142, in call
    acid.command(payload, handlers)
TypeError: command() missing 1 required positional argument: 'url'
```

**Comment by hkupty** [comment]
Can you check if you're using the latest version? I assumed the last commit would change line 142 to add the url parameter...

**Comment by daveyarwood** [comment]
Whoops! My apologies. I had removed it from my vimrc and then put it back, which _felt_ like a reinstall, but I forgot that the (not updated) repo was still in my `~/.vim/bundle/` directory.

I've updated acid.nvim, and now I get a different error when I run `:AcidBootstrap`:

```
error caught in async handler '/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid:command:AcidCommand [['Require']]'
Traceback (most recent call last):
  File "/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid/__init__.py", line 138, in acid_command
    command.call(self, self.context(), *args)
  File "/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid/commands/__init__.py", line 142, in call
    url = formatted_localhost_address(self.nvim)
NameError: name 'formatted_localhost_address' is not defined
```

**Comment by hkupty** [comment]
No worries. I'll fix this ASAP. 

I've grown to hate python for nvim plugins now that lua is embedded so I want to get rid of this as fast as I can.

**Comment by hkupty** [comment]
@daveyarwood Care to try now after this commit? I'm not properly able to retest acid due to the mods I'm making on the lua end...

**Comment by daveyarwood** [comment]
Updated again, now I get this upon `:AcidBootstrap`:

```
error caught in async handler '/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid:command:AcidCommand [['Require']]'
Traceback (most recent call last):
  File "/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid/__init__.py", line 138, in acid_command
    command.call(self, self.context(), *args)
  File "/home/dave/.vim/bundle/acid.nvim/rplugin/python3/acid/commands/__init__.py", line 144, in call
    url = formatted_localhost_address(self.nvim)
NameError: name 'self' is not defined
```

**Comment by daveyarwood** [comment]
I think I might be inclined to just try it again after acid.nvim is depythonified and enluafied.

**Comment by hkupty** [comment]
Sounds a good idea, as it'll put some pressure on me to make lua happen earlier. :)

**Comment by kuchichan** [comment]
Hi, thanks for reaction. Today I have reproduced the problem (pls look at "no REPL open"). Thanks in the advance, 
Cheers, Paweł
![acid_problem](https://user-images.githubusercontent.com/19415926/50310808-7cb61980-04a3-11e9-8fc5-0f138c4f6ac8.png)

**Comment by hkupty** [comment]
I see... I really messed with the handling. It should be simpler to revert to a few commits back then, and then I have more time to investigate further the issue and plan a better migration to the lua-based implementation.

Cheers,
Henry

**Comment by hkupty** [comment]
This last one should've fixed back. If this one doesn't work, I'll revert the commits.

Cheers and happy holidays

**Comment by kuchichan** [comment]
Hi, still I have the same problem, in the other hand there is a huge chance, that I messed up something.

**Comment by devth** [comment]
I haven't been able to get it working. Installed latest from master at `9e4bc44 2018-12-21` today.

```
nrepl 0.5.3
refactor-nrepl 2.4.0
cider-nrepl 0.19.0
leiningen 2.8.3
java 11
```

`checkhealth` shows:

```
## Python 3 provider (optional)
  - INFO: Using: g:python3_host_prog = "/usr/local/bin/python3"
  - INFO: Executable: /usr/local/bin/python3
  - INFO: Python version: 3.7.1
  - INFO: pynvim version: 0.3.1
  - OK: Latest pynvim is installed.
```

`UpdateRemotePlugins` shows:

```
|| remote/host: python3 host registered plugins ['acid']
```

Steps:

1. Open nvim
1. Start a terminal inside nvim and run `lein repl`
1. Once `lein repl` is fully loaded open a clj file
1. Says `No matching autocommands` in the status window
1. `car` also results in  `No matching autocommands` in the status window

If I try `<S-K>` on a symbol it errors with:

```
|| NvimError(b'Vim:E5300: Expected a Number or a String')
|| Traceback (most recent call last):
||   File "/Users/thartman/.config/nvim/plugged/acid.nvim/rplugin/python3/acid/handlers/__init__.py", line 82, in <lambda>
||     nvim.async_call(lambda: on_handle(msg, wc, key))
||   File "/Users/thartman/.config/nvim/plugged/acid.nvim/rplugin/python3/acid/handlers/doc.py", line 31, in on_handle
||     buf_win_nr = self.nvim.funcs.bufwinnr(self.doc_buf_nr)
||   File "/Users/thartman/Library/Python/3.7/lib/python/site-packages/pynvim/api/nvim.py", line 299, in call
||     return self.request('nvim_call_function', name, args, **kwargs)
||   File "/Users/thartman/Library/Python/3.7/lib/python/site-packages/pynvim/api/nvim.py", line 182, in request
||     res = self._session.request(name, *args, **kwargs)
||   File "/Users/thartman/Library/Python/3.7/lib/python/site-packages/pynvim/msgpack_rpc/session.py", line 102, in request
||     raise self.error_wrapper(err)
|| pynvim.api.nvim.NvimError: b'Vim:E5300: Expected a Number or a String'
||  
|| the call was requested at
||   File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 885, in _bootstrap
||     self._bootstrap_inner()
||   File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 917, in _bootstrap_inner
||     self.run()
||   File "../../.config/nvim/plugged/acid.nvim/rplugin/python3/acid/nrepl/__init__.py", line 68, in run
||     callback(incoming, self, key)
||   File "/Users/thartman/.config/nvim/plugged/acid.nvim/rplugin/python3/acid/handlers/__init__.py", line 82, in handler
||     nvim.async_call(lambda: on_handle(msg, wc, key))
```

**Comment by ttiurani** [comment]
This projects looks like my best bet on getting Clojure programming to work smoothly on neovim, so thanks for the effort so far!

I also had quite a few problems with running acid.nvim for the first time. I did manage to get nrepl connection to work (when I starting with some non-clojure file and only after that opened a .clj file), but then got:

"E116: Invalid arguments for function remote#define#CommandBootstrap" 

when trying to run evaluate on `(println "hello")`. The fix was to start instead the evaluation with single quotes `(println 'hello')` and only after that going back to double quotes, which then magically worked. Something needs one non-double-quoted eval to warm up, before it can handle double quotes :) The fix was found here: https://github.com/mikepjb/ice.nvim/blob/master/README.md so it seems quite old.

Hope you can find time to continue on the lua'tization of this project!

**Comment by hkupty** [comment]
Thanks @ttiurani.

I'll focus into it a bit more next week. Hopefully I'll be able to get everything back to normality fast enough to focus 100% on the lua port.

Cheers!

**Comment by hkupty** [comment]
Hopefully this issue is now obsolete as the port to lua is complete.

Let me know so I can close this :slightly_smiling_face:

**Comment by daveyarwood** [comment]
I was able to reinstall acid.nvim, run a REPL in another terminal, and then do `:AcidRequire` and then start eval-ing forms with `cpp`. I'm pleased with how lightweight it feels compared to fireplace, and the inline eval results display is awesome! Nice work! I'd say this issue can be closed. :tada: 

I am having an issue with using `K` to display docs -- it's throwing an error. I'll open a new issue for that.
